### PR TITLE
bitcoin: 0.20.1 -> 0.21.0 + allow build without wallet

### PR DESCRIPTION
**Pull request by prusnak**
###### Motivation for this change

* version update
* allow to build bitcoin{,d} without wallet via `--disable-wallet` configure option

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/blockchains/bitcoin.nix [hunk 1]
```diff
@@ -58,6 +60,8 @@ stdenv.mkDerivation rec {
   ] ++ optionals (!doCheck) [
     "--disable-tests"
     "--disable-gui-tests"
+] ++ optionals (!withWallet) [
```
Comment (roconnor) [inline pkgs/applications/blockchains/bitcoin.nix:63]: This looks improperly indented.

File: pkgs/applications/blockchains/bitcoin.nix [hunk 2]
```diff
@@ -36,15 +37,16 @@ stdenv.mkDerivation rec {
       "https://bitcoincore.org/bin/bitcoin-core-${version}/bitcoin-${version}.tar.gz"
       "https://bitcoin.org/bin/bitcoin-core-${version}/bitcoin-${version}.tar.gz"
     ];
-    sha256 = "4bbd62fd6acfa5e9864ebf37a24a04bc2dcfe3e3222f056056288d854c53b978";
+    sha256 = "1a91202c62ee49fb64d57a52b8d6d01cd392fffcbef257b573800f9289655f37";
```
Comment (roconnor) [inline pkgs/applications/blockchains/bitcoin.nix:40]: This hash matches the value signed by `Bitcoin Core binary release signing key` at https://bitcoincore.org/bin/bitcoin-core-0.21.0/SHA256SUMS.asc and I have verified the signature in that file.

File: pkgs/applications/blockchains/bitcoin.nix (hunk 1 above)
Comment (prusnak) [reply pkgs/applications/blockchains/bitcoin.nix]: Indent fixed in 672607c41f28570ea435989ac9f470e528a062ff

**Comment by roconnor** [comment]
We need to add an sqlite dependency (when wallets are enabled) for descriptor wallet support.

**Comment by prusnak** [comment]
> We need to add an sqlite dependency (when wallets are enabled) for descriptor wallet support.

Added in 727d4a5d5713408adece68915269fdaf2c901e03

**Comment by roconnor** [comment]
ping @jb55 in case he wants to also do a review.

**Comment by roconnor** [comment]
I'm not sure if we should add `? null` to the `db48` and `sqlite` parameters.  In practice it probably doesn't matter whether we do or not.

**Comment by erikarvstedt** [comment]
@prusnak, can you squash `bitcoin: add sqlite to buildInputs (if wallet is enabled)` into `bitcoin: 0.20.1 -> 0.21.0`?
This makes it clear that this is a new requirement for `0.21.0`.

**Comment by prusnak** [comment]
> I'm not sure if we should add `? null` to the `db48` and `sqlite` parameters. In practice it probably doesn't matter whether we do or not.

I don't see that as necessary.

**Comment by prusnak** [comment]
> can you squash `bitcoin: add sqlite to buildInputs (if wallet is enabled)` into `bitcoin: 0.20.1 -> 0.21.0`?

Done in f324c9a0b616bc0cb48ddded1744bfc66129346f

**Review by erikarvstedt — APPROVED** [review]
Tests `bitcoind`, `blockbook-frontend` succeed.
All other relevant tests (`prometheus-exporters`, `wasabibackend`) were already broken before.

**Review by roconnor — APPROVED** [review]
LGTM.

Someday we should look into perhaps not-wrapping non-qt binaries.

**Pull request merged**